src/server/transaction.cc

```diff
@@ -631,6 +632,8 @@ void Transaction::UnlockMulti() {
     sharded_keys[sid].push_back(k_v);
   }
 
+  SetMultiUniqueShardCount();
+
```

Comment:
Maybe we want to do this only if we have a journal open - its seems like this is an expensive operation.